From 453442a736a2ee938a48dcd22c0b62dcfc085d82 Mon Sep 17 00:00:00 2001
From: Philip Zeyliger <philip@cloudera.com>
Date: Tue, 15 Dec 2015 20:45:43 +0000
Subject: [PATCH 1040/2748] CLOUDERA-BUILD Exposing maven variables to ant

Reason: New Feature
Author: Philip Zeyliger
Ref: CDH-34616

To make repository.cloudera.com configurable,
it's necessary to expose Maven variables to
ant. This was already being done to an extent,
with ant calling 'mvn process-resources',
and a custom Maven plugin (com.cloudera.kitchen.maven-deps-to-props-plugin)
materializing a file with assorted CDH component and
dependency versions.  This approach, however,
did not immediately let me materialize
a Maven property, nor did I wish to use "mvn help:evaluate",
which is awkward and hard-to-parse.

Instead, I use Maven's standard "maven-resources-plugin"
to capture the variables I need.  More to the point,
once I'm using that, I can actually port over the existing
dependency version stuff straight to this built-in plugin,
reducing our world's plugin footprint by one.

I did the same thing in the Solr build earlier,
and it worked there.

To verify this, I simply ran diff.  Very astute readers
will notice that https://repository used to be https\://repository.
My reading of http://docs.oracle.com/javase/6/docs/api/java/util/Properties.html#load(java.io.Reader) as well as experimentation suggests
that it's not strictly necessary to escape that colon, and it works
fine, so I've left it.  (The new lines are ugly because I sorted
the input; they're nicely spaced in the real output.)

    $diff -u <(cat /mnt/philip/cdh2/repos/cdh5/hadoop/hadoop-mapreduce1-project/build.properties | sed -e s,5.7.0,5.7.6, | sort | uniq) <(cat build.p
    roperties | sort | uniq)
    --- /dev/fd/63  2015-12-15 20:46:44.531193170 +0000
    +++ /dev/fd/62  2015-12-15 20:46:44.531193171 +0000
    @@ -1,10 +1,20 @@
    -#
    +
     avro.version=1.7.6-cdh5.7.6-SNAPSHOT
    +# Default reactor.repo for developer builds.
    +# Defined canonically in the CDH root pom,
    +# do-component-build with the local .m2 repository.
     hadoop-common.version=2.6.0-cdh5.7.6-SNAPSHOT
     hadoop-hdfs.version=2.6.0-cdh5.7.6-SNAPSHOT
     hadoop-mapreduce.version=2.6.0-cdh5.7.6-SNAPSHOT
     hadoop.version=2.6.0-mr1-cdh5.7.6-SNAPSHOT
    -reactor.repo=https\://repository.cloudera.com/content/repositories/snapshots
    -#Tue Dec 15 20:30:15 UTC 2015
    +# Pakaging builds tend to override this in
    +reactor.repo=https://repository.cloudera.com/content/repositories/snapshots
    +releases.cloudera.com=file:///mnt/philip/local-repository/cdh-staging-local/
    +repository.root=file:///mnt/philip/local-repository
    +snapshots.cloudera.com=file:///mnt/philip/local-repository/cdh-snapshot-local/
    +# The default (at last check) is http://maven.jenkins.cloudera.com:8081/artifactory
    +# These override the settings in ivysettings.xml
    +# this may be overridden by ~/.m2/settings.xml.
     version=2.6.0-mr1-cdh5.7.6-SNAPSHOT
    +# Versions from root CDH pom.xml
     zookeeper.version=3.4.5-cdh5.7.6-SNAPSHOT

Change-Id: I473e31c422594dce62641e09baaa18f761a4141a
---
 hadoop-mapreduce1-project/build.xml                |    4 +++
 hadoop-mapreduce1-project/cloudera-pom.xml         |   26 +++++++++++++-------
 .../maven-packaging/templates/build.properties     |   22 +++++++++++++++++
 .../maven-packaging/templates/build.template       |    8 ------
 4 files changed, 43 insertions(+), 17 deletions(-)
 create mode 100644 hadoop-mapreduce1-project/cloudera/maven-packaging/templates/build.properties
 delete mode 100644 hadoop-mapreduce1-project/cloudera/maven-packaging/templates/build.template

diff --git a/hadoop-mapreduce1-project/build.xml b/hadoop-mapreduce1-project/build.xml
index 87ce353..a0c7156 100644
--- a/hadoop-mapreduce1-project/build.xml
+++ b/hadoop-mapreduce1-project/build.xml
@@ -2224,4 +2224,8 @@
 
   <!-- end of task-controller targets -->
 
+  <!-- Debug target to show properties. -->
+  <target name="list-all-properties">
+    <echoproperties />
+  </target>
 </project>
diff --git a/hadoop-mapreduce1-project/cloudera-pom.xml b/hadoop-mapreduce1-project/cloudera-pom.xml
index 17e0477..c8c9a0f 100644
--- a/hadoop-mapreduce1-project/cloudera-pom.xml
+++ b/hadoop-mapreduce1-project/cloudera-pom.xml
@@ -188,20 +188,28 @@ NOTE - This POM is used for Cloudera's build and packaging process.
     <plugins>
       
       <plugin>
-        <groupId>com.cloudera.kitchen</groupId>
-        <artifactId>maven-deps-to-props-plugin</artifactId>
+        <artifactId>maven-resources-plugin</artifactId>
+        <version>2.7</version>
         
         <executions>
           <execution>
-            <id>build.properties</id>
-            <configuration>
-              <templateFile>${basedir}/cloudera/maven-packaging/templates/build.template</templateFile>
-              <targetFile>${basedir}/build.properties</targetFile>
-            </configuration>
+            <id>copy-resources</id>
+            <phase>process-resources</phase>
             <goals>
-              <goal>generate</goal>
+              <goal>copy-resources</goal>
             </goals>
-            <phase>process-resources</phase>
+            <configuration>
+              <outputDirectory>${basedir}</outputDirectory>
+              <resources>
+                <resource>
+                  <directory>${basedir}/cloudera/maven-packaging/templates</directory>
+                  <includes>
+                    <include>build.properties</include>
+                  </includes>
+                  <filtering>true</filtering>
+                </resource>
+              </resources>
+            </configuration>
           </execution>
         </executions>
       </plugin>
diff --git a/hadoop-mapreduce1-project/cloudera/maven-packaging/templates/build.properties b/hadoop-mapreduce1-project/cloudera/maven-packaging/templates/build.properties
new file mode 100644
index 0000000..96f1cba
--- /dev/null
+++ b/hadoop-mapreduce1-project/cloudera/maven-packaging/templates/build.properties
@@ -0,0 +1,22 @@
+# Defined canonically in the CDH root pom,
+# this may be overridden by ~/.m2/settings.xml.
+# The default (at last check) is http://maven.jenkins.cloudera.com:8081/artifactory
+repository.root=${repository.root}
+
+# These override the settings in ivysettings.xml
+snapshots.cloudera.com=${repository.root}/cdh-snapshot-local/
+releases.cloudera.com=${repository.root}/cdh-staging-local/
+
+# Default reactor.repo for developer builds.
+# Pakaging builds tend to override this in
+# do-component-build with the local .m2 repository.
+reactor.repo=https://repository.cloudera.com/content/repositories/snapshots
+
+# Versions from root CDH pom.xml
+version=${cdh.mr1.version}
+hadoop.version=${cdh.mr1.version}
+hadoop-common.version=${cdh.hadoop.version}
+hadoop-hdfs.version=${cdh.hadoop.version}
+hadoop-mapreduce.version=${cdh.hadoop.version}
+avro.version=${cdh.avro.version}
+zookeeper.version=${cdh.zookeeper.version}
diff --git a/hadoop-mapreduce1-project/cloudera/maven-packaging/templates/build.template b/hadoop-mapreduce1-project/cloudera/maven-packaging/templates/build.template
deleted file mode 100644
index 5125c23..0000000
--- a/hadoop-mapreduce1-project/cloudera/maven-packaging/templates/build.template
+++ /dev/null
@@ -1,8 +0,0 @@
-version=org.apache.hadoop:hadoop-core.version
-hadoop.version=org.apache.hadoop:hadoop-core.version
-hadoop-common.version=org.apache.hadoop:hadoop-common.version
-hadoop-hdfs.version=org.apache.hadoop:hadoop-hdfs.version
-hadoop-mapreduce.version=org.apache.hadoop:hadoop-mapreduce.version
-reactor.repo=https://repository.cloudera.com/content/repositories/snapshots
-avro.version=org.apache.avro:avro.version
-zookeeper.version=org.apache.zookeeper:zookeeper.version
-- 
1.7.9.5

